pkg/workspace/library.go

```diff
@@ -120,33 +98,96 @@ func (l *Library) FindFile(path string) (*files.File, error) {
 	return nil, fmt.Errorf("Expected to find file %s", path)
 }
 
-type FileInLibrary struct {
-	File            *files.File
-	Library         *Library
-	parentLibraries []*Library
+type PackageInLibrary struct {
+	pkgPath []*Library
 }
 
-func (fileInLib *FileInLibrary) RelativePath() string {
+func (packageInLib *PackageInLibrary) Package() *Library {
+	return packageInLib.pkgPath[len(packageInLib.pkgPath)-1]
+}
+
+func (packageInLib *PackageInLibrary) Library() *Library {
+	return packageInLib.pkgPath[0]
+}
+
+var LoaderRegexp = regexp.MustCompile("^((@?)([^:]*):)?(.*)$")
```

Comment:
should use \A...\z instead of ^...$

Reply:
lets also make a dedicated class LoadPath that has Library(), Package(), Module() accessors. reading code below that uses matches[x] is quite tedious. we should also add some unit tests that test out different variations.